### ISSUE TITLE
Calling MakeADFun in a loop seems to cause a crash?

### ISSUE THREAD

**Issue by alexfun** [kaskr/adcomp#232]
Sorry, this is a very vague question but I am wondering whether anyone else has encountered something like this.

I have an objective function `obj_fun.cpp` compiled to be `obj_fun.dll`, and loaded using `dyn.load`. I then build a loop that looks like this in `R`:

```
for (i = 1:big_integer) {
reg_data <- list(...)
param_data <- list(...)
optim_obj <- MakeADFun(reg_data, param_data, DLL = 'obj_fun.dll')
.... do non-linear constrained optimisation using nloptr::mma ...
}
```

The loop runs fine most of the time, but sometimes at some big `i` it will cause the underlying R session to crash, with no errors printed to `RStudio`.  (Windows does throw an error message regarding the dll but I forgot to screenshot it).

What is really weird to me is that after restarting the R session, and restarting the loop from scratch (i = 1), or at the index `i` where the crash occurred is usually sufficient to make the loop run fine. As far as I can tell everything is deterministic except possibly in the optimisation step (where the path to the optimal solution may be slightly random?).

**Comment by kaskr** [comment]
Sounds nasty. Recently some bugs were fixed ( a046a5ef9657bfa2853c4c573d4bc1693af190b6 ) that could in theory cause this issue if your template uses the `REPORT` macro. Which TMB version are you using ?

**Comment by alexfun** [comment]
Thanks. Yes, I do use the report macro. My version is TMB_1.7.8. It looks like the fix is in 1.7.10 so I should upgrade. I have found that unloading and then loading the DLL after a batch of runs, before starting the next run has eliminated the behaviour with crashes.
